src/main/java/de/tum/in/www1/artemis/config/migration/entries/BambooMigrationService.java

```diff
@@ -0,0 +1,499 @@
+        if (credentialsId.isEmpty()) {
+            log.error("No shared credential found for git user " + gitUser + ". Migration will fail.");
+            throw new RuntimeException("No shared credential found for git user " + gitUser
+                    + " in Bamboo. Migration will fail. Please create a shared Username and Password credential for this user and run " + "the migration again.");
+        }
+        this.sharedCredentialId = credentialsId;
+    }
+
+    @Override
+    public void overrideBuildPlanRepository(String buildPlanId, String name, String repositoryUrl, String defaultBranch) {
+        if (this.sharedCredentialId.isEmpty()) {
+            Optional<Long> credentialsId = getSharedCredential();
+            if (credentialsId.isEmpty()) {
+                log.error("No shared credential found for git user " + gitUser + ". Migration will fail.");
+            }
+            this.sharedCredentialId = credentialsId;
+        }
+        Optional<Long> repositoryId = getConnectedRepositoryId(buildPlanId, name);
+
+        if (repositoryId.isEmpty()) {
+            log.info("Repository " + name + " not found for build plan " + buildPlanId + ", will be added now");
+        }
+        else {
+            log.debug("Deleting repository " + name + " for build plan " + buildPlanId);
+            deleteLinkedRepository(buildPlanId, repositoryId.get());
+        }
+        log.debug("Adding repository " + name + " for build plan " + buildPlanId);
+        addGitRepository(buildPlanId, bambooInternalUrlService.toInternalVcsUrl(repositoryUrl), name, this.sharedCredentialId.orElseThrow(), defaultBranch);
+    }
+
+    private List<Long> getAllTriggerIds(String buildPlanName) {
+        if (buildPlanName == null) {
+            return List.of();
+        }
+        MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
+        parameters.add("buildKey", buildPlanName);
+        String requestUrl = bambooServerUrl + "/chain/admin/config/editChainTriggers.action";
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(requestUrl).queryParams(parameters);
+
+        var response = restTemplate.exchange(builder.build().toUri(), HttpMethod.GET, null, String.class);
+        return getDataItemIds(response.getBody());
+    }
+
+    /**
+     * Returns the id of the repository with the given name for the given build plan.
+     * If no repository with the given name exists, an empty optional is returned.
+     *
+     * @param buildPlanId The key of the build plan, e.g. 'EIST16W1-BASE'.
+     * @param name        The name of the repository, e.g. 'assignment'.
+     * @return the id of the repository with the given name for the given build plan
+     */
+    private Optional<Long> getConnectedRepositoryId(String buildPlanId, String name) {
+        if ((name == null || name.isEmpty()) || (buildPlanId == null || buildPlanId.isEmpty())) {
+            return Optional.empty();
+        }
```

Comment:
I'd rather keep it in this method because it will clutter the calling methods too much to check it there. This case should not happen and this is just safeguarding to prevent erroneous calls to Bamboo.

Reply:
If it will never happen, should we add an error log here?